bob/src/cluster/quorum.rs

```diff
@@ -318,6 +319,93 @@ impl Quorum {
         }
         Ok(())
     }
+
+    async fn collect_remote_exists(
+        result: &mut [bool],
+        keys: &[BobKey],
+        indexes_by_node: &mut HashMap<Node, IndexMap>,
+    ) {
+        if indexes_by_node.is_empty() {
+            return;
+        }
+
+        let mut node_keys_by_node_name = HashMap::new();
+        for (node, node_map) in indexes_by_node.iter_mut() {
+            node_map.retain_not_existed(&result);
+            if !node_map.is_empty() {
+                node_keys_by_node_name.insert(node.name().to_owned(), (node.clone(), node_map.collect(keys)));
+            }
+        }
+
+        if !node_keys_by_node_name.is_empty() {
+            let nodes: Vec<_> = node_keys_by_node_name.values().map(|(n, _)| n).collect();
```

Comment:
No need to collect nodes. `exist_on_remote_nodes` can use `Node` instances from `node_keys_by_node_name`